sys/newlib/syscalls.c

```diff
@@ -200,15 +204,21 @@ int _open_r(struct _reent *r, const char *name, int mode)
  */
 int _read_r(struct _reent *r, int fd, void *buffer, unsigned int count)
 {
-#ifndef MODULE_UART0
-    while (rx_buf.avail == 0) {
-        mutex_lock(&uart_rx_mutex);
-    }
-    return ringbuffer_get(&rx_buf, (char*)buffer, rx_buf.avail);
-#else
+    (void) r;
+    (void) fd;
+#ifdef MODULE_UART0
+    (void) count;
     char *res = (char*)buffer;
     res[0] = (char)uart0_readc();
     return 1;
+#else
+    while (rx_buf.avail == 0) {
+        mutex_lock(&uart_rx_mutex);
+    }
+    if (count > rx_buf.avail) {
+        count = rx_buf.avail;
+    }
```

Comment:
yeah, sorry, I thought I made a separate PR of that commit. https://github.com/gebart/RIOT/commit/a22971e049936b81315f8315095078ead0d7d3bf

It's a fix for a potential buffer overflow.